### ISSUE TITLE
Watcher webhook actions returning 302 response code in 7.2 and above.

### ISSUE THREAD

**Issue by jamesspi** [elastic/elasticsearch#44970]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.2, 7.3

**Plugins installed**: []

**JVM version** (`java -version`): Bundled JDK

**OS version** (`uname -a` if on a Unix-like system): ESS and Darwin Kernel Version 18.6.0

**Description of the problem including expected versus actual behavior**: Watcher web hook actions do not return a 200 status code for POST requests.

**Steps to reproduce**: Execute a watch with a webhook action.

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Create watch with a webhook action.
 2. Force execute watch, or wait for it to trigger

As of `7.2`, webhook actions are are returning a `302` http status code for POST requests, as opposed to `200`. After some initial discussions with @jakelandis, there is nothing obvious in the codebase that would cause this.

The same exact webhook action in `7.1` and below works fine. All other http clients (tested with curl, wget and postman) all return the expected response. 

This has been tested (an initially spotted) in ESS, and also on my MBP.

**Comment by spinscale** [comment]
can you provide a reproducible example here? I tried this and this properly returns 200, so there must be something else that you are doing

```
POST _watcher/watch/_execute
{
  "watch": {
    "trigger": {
      "schedule": {
        "interval": "10h"
      }
    },
    "input": {
      "http": {
        "request": {
          "url" : "https://httpbin.org/status/200"
        }
      }
    },
    "actions": {
      "logme": {
        "logging": {
          "text": "{{ctx.payload._status_code}}"
        }
      }
    }
  }
}
```

I also tried `https://www.elastic.co/` as an URL, and that one also returned the proper error code.

**Comment by jamesspi** [comment]
Hey @spinscale ,

Here is one of the watches:

```
{
  "trigger": {
    "schedule": {
      "interval": "1m"
    }
  },
  "input": {
    "search": {
      "request": {
        "search_type": "query_then_fetch",
        "indices": [
          "ct-*"
        ],
        "rest_total_hits_as_int": true,
        "body": {
          "query": {
            "bool": {
              "must": [
                {
                  "query_string": {
                    "query": "dns_name:*.swiftcrypto.com AND log_entry_stage.keyword :\"Certificate\""
                  }
                },
                {
                  "range": {
                    "@timestamp": {
                      "gte": "now-1m"
                    }
                  }
                }
              ]
            }
          },
          "sort": [
            {
              "@timestamp": {
                "order": "desc"
              }
            }
          ]
        }
      }
    }
  },
  "condition": {
    "compare": {
      "ctx.payload.hits.total": {
        "gt": 0
      }
    }
  },
  "actions": {
    "stdlib": {
      "webhook": {
        "scheme": "https",
        "host": "jamesspi.lib.id",
        "port": 443,
        "method": "post",
        "path": "/swiftcrypto-sec-bot@dev/ct/",
        "params": {},
        "headers": {
          "Content-Type": "application/x-www-form-urlencoded"
        },
        "body": "dnsname={{#ctx.payload.hits.hits}}{{_source.dns_name}},{{/ctx.payload.hits.hits}}&title=New%20Certificate%20Detected&title_url=https://ffa0e3abcb444c9ab93ffcefcbdb60a8.europe-west1.gcp.cloud.es.io:9243/app/kibana#/doc/2ecb6bd0-1d85-11e9-85c5-335acafba1db/ct-*/doc?id={{ctx.payload.hits.hits.0._id}}&_g=()&text={{ctx.payload.hits.total}}%20New%20certificates%20have%20been%20created%20according%20to%20certificate%20transparency%20logs%20in%20the%20last%20minute.&issuer={{#ctx.payload.hits.hits}}{{_source.issuer_organisation}},{{/ctx.payload.hits.hits}}&entry={{#ctx.payload.hits.hits}}{{_source.log_entry_url}},{{/ctx.payload.hits.hits}}&main_text=SwiftCrypto%20Alert%20"
      }
    }
  },
  "throttle_period_in_millis": 3000
}
```

As described, it is only returning `302` for `POST` requests. In your example, you're making a `GET` request. `GET` is still working as expected.

Here is a doc if you want to try:

```
{
    "Pubkey": "d9a7713ba8ea233f3580feade68be568d7246baba8cf540022a365c3f600a687",
    "log_entry_id": "673984766",
    "@timestamp": "2019-08-04T01:10:04.016Z",
    "log_entry_stage": "Certificate",
    "dns_name": "loader.swiftcrypto.com",
    "sha256": "476719470ca9028146cd5831623bf2eb149e71b0f12dee5aa048852b25f80b63",
    "certificate_expiry_date": "2019-11-02T00:08:00.000Z",
    "issuer_organisation": "Let's Encrypt",
    "certificate_generation_date": "2019-08-04T00:08:00.000Z",
    "issuer_country_code": "US",
    "issuer_common_name": "Let's Encrypt Authority X3",
    "log_entry_url": "https://ct.googleapis.com/logs/argon2019/"
  }
```

**Comment by spinscale** [comment]
I found the issue.. our own HTTPClient is creating a wrong URI, which then your endpoint creates another redirect for. This uncovered two bugs, which have never been uncovered because of the forgiveness of many endpoints it seems to me.

Should have a PR up shortly.

**Comment by jamesspi** [comment]
@spinscale did the fix make it to `7.3.1`? I am still having the same issue.

**Comment by spinscale** [comment]
this is still an issue, as the test assumptions are wrong... I'll have a fix ready soon.

**Comment by jamesspi** [comment]
@spinscale this is still persistent in `7.3.2`.

**Comment by spinscale** [comment]
the fix has been backported to 7.4.0 and above only (the labels on the PR the correct ones)

**Comment by jamesspi** [comment]
@spinscale ah, my bad. Thanks!
